### PR TITLE
Allowing for templates, fixed one bug, making validators longer

### PR DESCRIPTION
**Pull request by ghinch**
3 changes:
1) fixed a bug referring to 'signers' rather than 'signer'
2) added a method for creating a document from a template
3) increased the name/email validator length to 255, as that is the valid length in the API and sometimes you need it. Also made email required, because well it is.

### CODE REVIEW

File: hellosign/hellosign.py [hunk 1]
```diff
@@ -32,7 +32,7 @@ def add_signer(self, signer):
             self.signers.append(signer)
         else:
             if not signer.validate():
-                raise Exception("HelloSigner Errors %s" % (signers.errors,))
```
Comment (rosscdh) [inline hellosign/hellosign.py:35]: Good catch. forest for the trees in this case.

File: hellosign/hellosign.py [hunk 2]
```diff
@@ -90,10 +90,28 @@ def create(self, *args, **kwargs):
             auth = kwargs['auth']
             del(kwargs['auth'])
 
-        self.validate()
+        self.validate_signers()
```
Comment (rosscdh) [inline hellosign/hellosign.py]: I like this separation implementation but kinda prefer if the self.validate() called them individually. Thoughts?

**Comment by rosscdh** [comment]
Just FYI at the moment, this library as well as its sister https://github.com/rosscdh/django-hello_sign is in a state of flux; we are rapidly building it out to suit our needs. But feel free to get stuck in too.

Regards
Ross

File: hellosign/hellosign.py (hunk 2 above)
Comment (ghinch) [inline hellosign/hellosign.py]: Don't see any problem with that, sure

**Comment by ghinch** [comment]
Oh and if possible @isnotahippy should share credit on these changes, I had to pull them from our fork which lost his commits it seems in the process, but he worked on this as well.

**Comment by rosscdh** [comment]
@ghinch @isnotahippy added authors https://github.com/rosscdh/hellosign/blob/master/README.md

**Comment by rosscdh** [comment]
@ghinch @isnotahippy just to let you guys know. as of yesterday HelloSign have released their official python sdk found at https://github.com/HelloFax/hellosign-python-sdk

**Pull request merged**